dbuild/main.d

```diff
@@ -0,0 +1,249 @@
+core/sync/
+core/sys/netbsd/
+core/sys/bionic/
+
+## excluded files
+core/sys/darwin/dlfcn.d
+core/sys/linux/stdio.d
+core/sys/osx/sys/event.d
+core/sys/posix/mqueue.d
+core/sys/posix/sys/msg.d
+
+`.splitEntries;
+
+    foreach (a; excluded)
+        if (file.startsWith(a))
+            return false;
+    return true;
+}
+
+// TODO: check why all those exclusions are needed or if it was a longstanding error
+bool isValid_src(string file)
+{
+    enum excluded = `
+## excluded dirs
+core/sys/netbsd/
+core/sys/bionic/
+core/sys/openbsd/
+core/sys/osx/
+core/sys/darwin/mach/
+core/sys/solaris/
+core/stdcpp/
+
+## excluded files ; TODO: is there some logic/pattern here (to auto-generate)
+core/stdc/wctype.d
+core/stdc/tgmath.d
+core/sys/darwin/dlfcn.d
+core/sys/darwin/execinfo.d
+core/sys/darwin/pthread.d
+core/sys/darwin/sys/cdefs.d
+core/sys/darwin/sys/event.d
+core/sys/darwin/sys/mman.d
+core/sys/freebsd/pthread_np.d
+core/sys/linux/config.d
+core/sys/linux/dlfcn.d
+core/sys/linux/elf.d
+core/sys/linux/epoll.d
+core/sys/linux/errno.d
+core/sys/linux/execinfo.d
+core/sys/linux/fcntl.d
+core/sys/linux/ifaddrs.d
+core/sys/linux/link.d
+core/sys/linux/sched.d
+core/sys/linux/sys/auxv.d
+core/sys/linux/sys/eventfd.d
+core/sys/linux/sys/file.d
+core/sys/linux/sys/netinet/tcp.d
+core/sys/linux/sys/prctl.d
+core/sys/linux/termios.d
+core/sys/linux/time.d
+core/sys/linux/timerfd.d
+core/sys/linux/unistd.d
+core/sys/posix/config.d
+core/sys/posix/dlfcn.d
+core/sys/posix/fcntl.d
+core/sys/posix/grp.d
+core/sys/posix/iconv.d
+core/sys/posix/inttypes.d
+core/sys/posix/libgen.d
+core/sys/posix/mqueue.d
+core/sys/posix/net/if_.d
+core/sys/posix/netinet/tcp.d
+core/sys/posix/poll.d
+core/sys/posix/pthread.d
+core/sys/posix/pwd.d
+core/sys/posix/sched.d
+core/sys/posix/semaphore.d
+core/sys/posix/setjmp.d
+core/sys/posix/stdio.d
+core/sys/posix/stdlib.d
+core/sys/posix/sys/filio.d
+core/sys/posix/sys/ioccom.d
+core/sys/posix/sys/msg.d
+core/sys/posix/sys/ttycom.d
+core/sys/posix/syslog.d
+core/sys/posix/termios.d
+core/sys/posix/time.d
+core/sys/posix/ucontext.d
+core/sys/posix/unistd.d
+core/sys/posix/utime.d
+test_runner.d
+
+`.splitEntries;
+
+    auto added = `
+  core/sys/solaris/sys/priocntl.d
+  core/sys/solaris/sys/procset.d
+  core/sys/solaris/sys/types.d
+  `.splitEntries;
+
+    if (added.canFind(file))
+        return true;
+
+    foreach (a; excluded)
+        if (file.startsWith(a))
+            return false;
+    return true;
+}
+
+auto files_IMPORTS = `
+core/sync/barrier.di
+core/sync/condition.di
+core/sync/config.di
+core/sync/exception.di
+core/sync/mutex.di
+core/sync/rwmutex.di
+core/sync/semaphore.di
+  `.splitEntries.array;
+
+bool isValid_DOCS(string file)
+{
+    file = file.stripExtension.replace("/", "."); // FACTOR
+    import std.stdio;
+
+    if (file.startsWith("rt.", "core.stdc.", "core.stdcpp.", "core.sync."))
+        return true;
+
+    import std.regex;
+
+    if (matchFirst(file, `^(object|core\.\w+)$`.regex))
```

Comment:
with latest change it will compile just once (since it now uses rdmd) so this cost should not matter; in this case, keeping code short and idiomatic is more important.